rs/nns/cmc/src/main.rs

```diff
@@ -224,6 +227,16 @@ pub struct StateV1 {
 
     pub total_cycles_minted: Cycles,
 
+    // We use this for synchronization/journaling.
+    //
+    // Because our operations (e.g. minting cycles) require calling other
+    // canister(s), in particular ledger, it is possible for duplicate requests
+    // to interleave. In such cases, we want subsequent operations to see that
+    // an operation is already in flight. Therefore, before making any canister
+    // calls, we check that the block does not already have a status, and set
+    // its status to Processing. Only then do we proceed with calling the other
```

Comment:
Super helpful comment, thanks for adding it!

> Only then ...

Should is be mentioned what happens otherwise?